src/server/generic_family.cc

```diff
@@ -623,17 +625,12 @@ void OpScan(const OpArgs& op_args, const ScanOpts& scan_opts, uint64_t* cursor,
   // 10k Traverses
   const size_t limit = 10000;
   do {
+    cur = prime_table->Traverse(
+        cur, [&](PrimeIterator it) { cnt += ScanCb(op_args, it, scan_opts, vec); });
+    ++buckets_iterated;
     if (buckets_iterated >= limit) {
-      buckets_iterated = 0;
-      util::ThisFiber::Yield();
-    }
-    {
-      // Disable flush journal changes to prevent preemtion in traverse.
-      journal::JournalFlushGuard journal_flush_guard(op_args.shard->journal());
-      cur = prime_table->Traverse(
-          cur, [&](PrimeIterator it) { cnt += ScanCb(op_args, it, scan_opts, vec); });
+      break;
```

Comment:
I believe this can be inside the while condition

Reply:
You are right. I did not want to make the expression big so I opted in for an if statement :shrug: 

One sec will address this.